packages/storage/src/implementation/taskenums.ts

```diff
@@ -20,18 +20,41 @@
  */
 
 /**
- * Enum for task events.
+ * An event that is triggered on a task.
+ * @public
  */
 export type TaskEvent = string;
+
+/**
+ * An event that is triggered on a task.
+ * @public
+ */
 export const TaskEvent = {
-  /** Triggered whenever the task changes or progress is updated. */
+  /**
+   * For this event,
+   * <ul>
+   *   <li>The `next` function is triggered on progress updates and when the
+   *       task is paused/resumed with an `UploadTaskSnapshot` as the first
+   *       argument.</li>
+   *   <li>The `error` function is triggered if the upload is canceled or fails
+   *       for another reason.</li>
+   *   <li>The `complete` function is triggered if the upload completes
+   *       successfully.</li>
+   * </ul>
+   */
   STATE_CHANGED: 'state_changed'
 };
 
 /**
  * Internal enum for task state.
+ * @internal
  */
 export type InternalTaskState = string;
+
+/**
+ * Internal enum for task state.
+ * @internal
```

Comment:
It could be an enum but not const enum because it's imported elsewhere. It seems like you can't export const enums for use in another file but let me know if I'm just missing something.

I removed all tags in this file since there are public types for them.

Reply:
See my comment earlier.

Reply:
I was able to convert all 3 enums to const enum (the public ones have a string literal type in storage-types/exp so we don't need to export the ones here) but I had to move the function `taskStateFromInternalTaskState` out of the file (I moved it to `task.ts`, the only file that uses it).

Reply:
Restored, same as StorageErrorCode.